ktor-server/ktor-server-plugins/ktor-server-sse/common/src/io/ktor/server/sse/ServerSSESession.kt

```diff
@@ -130,3 +136,36 @@ public suspend inline fun <reified T : Any> ServerSSESessionWithSerialization.se
 public suspend inline fun <reified T : Any> ServerSSESessionWithSerialization.send(data: T) {
     send(ServerSentEvent(serializer(typeInfo<T>(), data)))
 }
+
+/**
+ * Starts a heartbeat for the ServerSSESession.
+ *
+ * The heartbeat will send the specified [Heartbeat.event] at the specified [Heartbeat.duration] interval
+ * as long as the session is active.
+ *
+ * @param heartbeatConfig a lambda that configures the [Heartbeat] object used for the heartbeat.
+ */
+public fun ServerSSESession.heartbeat(heartbeatConfig: Heartbeat.() -> Unit) {
+    val heartbeat = Heartbeat().apply(heartbeatConfig)
+    val heartbeatJob = Job()
+    launch(heartbeatJob) {
```

Comment:
Just wondering why we don't add `CoroutineName` to launched coroutines? In theory this could simplify further debugging. Both, for us and for users.
@marychatte, @bjhham, WDYT?